src/types.rs

```diff
@@ -264,6 +264,6 @@ impl BufRingEntry {
     /// a valid pointer type, but also the argument must be the address to the first entry
     /// of the buf_ring for the resv field to even be considered the tail field of the ring.
     pub unsafe fn tail(ring_base: *const BufRingEntry) -> *const u16 {
-        std::ptr::addr_of!((*ring_base).0.resv)
+        &(*ring_base).0.resv
```

Comment:
Is this unsound because it may create a valid reference to uninitialized memory?

Reply:
We know that the memory is zero-initialized. I don't know if that is enough to avoid UB according to the language's rules, though. We could get around that question by manually initializing the first `BufRingEntry` in the ring, in `InnerBufRing::new`.

In that case, do we want to do without this `tail` method completely?

Reply:
Maybe we should use `autocfg` so that we can use `addr_of!` on new compiler.

Reply:
I'm not sure, we'd still have to make sure that the code compatible with older compilers is correct, but then there isn't much point in having another variant of the same code doing the same thing.

Reply:
I believe that on old compiler it was probably more of a definition UB and didn't really cause problems. The point of using `addr_of!` is that old code will work if compiler behavior changes in future.

like https://github.com/Gilnaa/memoffset/blob/master/src/raw_field.rs#L21

I've thought about it, and a C repr type reference to a zero-fill buffer should be safe. but not using `addr_of!` means a stricter requirement on parameter.

Reply:
> I believe that on old compiler it was probably more of a definition UB and didn't really cause problems. The point of using `addr_of!` is that old code will work if compiler behavior changes in future.

Ah, I see.

> I've thought about it, and a C repr type reference to a zero-fill buffer should be safe. but not using `addr_of!` means a stricter requirement on parameter.

Is `BufRingEntry::tail` really useful to external users? If not, we could just drop it, given it is unstable.

Reply:
> Is `BufRingEntry::tail` really useful to external users? If not, we could just drop it, given it is unstable.

?